src/exceptions.jl

```diff
@@ -0,0 +1,22 @@
+struct RayTaskException <: Exception
+    task_name::String
+    pid::Int
+    ip::IPAddr
+    task_id::String
+    captured::CapturedException
+end
+
+function RayTaskException(task_name::AbstractString, captured::CapturedException)
+    return RayTaskException(task_name, getpid(), getipaddr(), get_task_id(), captured)
+end
+
+function Base.showerror(io::IO, ex::RayTaskException, bt=nothing; backtrace=true)
```

Comment:
>  I guess it's so that we can control when the backtrace is shown (i.e., _before_ showing the nested exceptions)?

Correct. 

A more detailed answer requires you consider these two methods:
```julia
function showerror(io::IO, ce::CapturedException)
    showerror(io, ce.ex, ce.processed_bt, backtrace=true)
end
```
– https://github.com/JuliaLang/julia/blob/57e70c1602c59d812f30ff5fbc992e52de3a4b13/base/task.jl#L24-L26

```julia
function showerror(io::IO, ex, bt; backtrace=true)
    try
        showerror(io, ex)
    finally
        backtrace && show_backtrace(io, bt)
    end
end
```
– https://github.com/JuliaLang/julia/blob/57e70c1602c59d812f30ff5fbc992e52de3a4b13/base/errorshow.jl#L93-L99

When showing a `CapturedException` the `showerror` method gets recursively called until all of the exception messages are displayed and then all of the stack traces will be printed. Due to the 3-arg `showerror` we have to specialize on this method so we can show exception messages and their stacktraces in this order.